pathfinder/model/token_network.py

```diff
@@ -190,30 +183,33 @@ def update_fee(
         else:
             raise ValueError('Signature does not match any of the participants.')
 
-        new_fee_casted = float(new_fee)
-        channel_view = self.G[sender][receiver]['view']
+        new_percentage_fee_casted = float(new_percentage_fee)
+        channel_view: ChannelView = self.G[sender][receiver]['view']
+
+        if nonce <= channel_view.fee_info_nonce:
+            raise ValueError('Outdated fee info.')
```

Comment:
I moved it here because the balance proof nonce is stored in the `ChannelView` as well. Doesn't fit perfectly, but I don't think having it in the PathfindingService would improve stuff much. 
Open for ideas...